File: domain/src/main/kotlin/models/Hotel.kt
package models

import hotelService.api.GeoLocation

data class Hotel(
    val id: Int,
    val name: String,
    val address: GeoLocation,
    val price: Double,
    val rating: Int,
    val images: HotelImages?
)

data class HotelImages(val mainImage: String, val extraImages: List<String>)

File: web/app/src/models/entity/Hotel.js
import { HotelImages } from "./HotelImages";

export class Hotel {
  id;
  name;
  address;
  price;
  rating;
  imageUrl;

  /**
   * @param {Number} id
   * @param {String} name
   * @param {Object} address
   * @param {Number} price
   * @param {Number} rating
   * @param {HotelImages} images
   */
  constructor(id, name, address, price, rating, images) {
    this.name = name;
    this.address = address;
    this.price = price;
    this.rating = rating;
    this.images = images;
    this.id = id;
  }
}

Hotel.fromObject = (obj) => {
  return new Hotel(
    obj.id,
    obj.name,
    obj.address,
    obj.price,
    obj.rating,
    HotelImages.fromObject(obj.images)
  );
};

Hotel.Null = new Hotel(0, "", {}, 0, 0, HotelImages.Null);
